packages/common/src/request.ts

```diff
@@ -17,34 +17,54 @@ export class RemoteServerError extends Error {
 }
 
 /**
+ * ```js
+ * import { hubApiRequest } from "@esri/hub-common";
+ * //
+ * hubApiRequest(
+ *   "/datasets",
+ *   requestOptions
+ * })
+ *   .then(response);
+ * ```
  * make a request to the Hub API
- * @param url URL to request
+ * @param route API route
  * @param requestOptions request options
  */
-export function hubRequest(url: string, requestOptions?: IHubRequestOptions) {
-  // TODO: use request() under the hood?
-  // requestOptions.params.f = null;
-  // return request(url, requestOptions);
-  // TODO: cast to JSONAPI document?
-  // .then(response => {
-  //   const { data, meta } = response;
-  //   return { data, meta } as Document;
-  // });
-  // TODO: base on requestOptions
-  const fetchFn = /* requestOptions.fetch || */ fetch;
-  return fetchFn(url, {
-    method: "POST", // TODO: get from requestOptions?
-    headers: {
-      // TODO: base on request requestOptions?
-      "Content-Type": "application/json"
-    }
-    // TODO: base on requestOptions.params?
-    // body: JSON.stringify(body)
-  }).then(resp => {
-    if (resp.ok) {
-      return resp.json();
-    } else {
-      throw new RemoteServerError(resp.statusText, url, resp.status);
-    }
+export function hubApiRequest(
+  route: string,
+  requestOptions?: IHubRequestOptions
+) {
+  // merge in default request options
+  const options: IHubRequestOptions = {
+    ...{
+      hubApiUrl: "https://opendata.arcgis.com/api/v3/",
+      httpMethod: "POST",
+      fetch
+    },
+    ...requestOptions
+  };
+  // merge in default headers
+  const headers = {
+    ...{ "Content-Type": "application/json" },
+    ...options.headers
+  };
+  // TODO: build query params/body based on requestOptions.params?
+  // build Hub API URL
+  const url = buildUrl({
+    host: options.hubApiUrl,
+    path: route
   });
+  return options
+    .fetch(url, {
+      method: options.httpMethod,
+      headers
+      // body: JSON.stringify(body)
```

Comment:
Is the commented-out line intentional?  Seems like it would prevent POST body from being sent.

Reply:
that's a TODO, I'm just implementing the bare minimum that I need to ship `getContent()`